torch/_inductor/sizevars.py

```diff
@@ -359,6 +359,11 @@ def evaluate_static_shape(self, left: Expr) -> int:
     def evaluate_static_shapes(self, left: List[Expr]) -> List[int]:
         return [self.evaluate_static_shape(x) for x in left]
 
+    def remove_precomputed_replacements(self, expr: Expr) -> Expr:
+        if any(s.name.startswith("ps") for s in expr.free_symbols):
+            return sympy_subs(expr, self.inv_precomputed_replacements)
+        return expr
```

Comment:
Previously, the removal was recursive until no `ps*` symbols remain to replace. Now it's not: there is just one replacement pass. I guess, if some of the expressions in the values of the `self.inv_precomputed_replacements` contain some of the symbols in the keys, one replacement pass won't be sufficient here?

Reply:
I changed `self.inv_precomputed_replacements` so that we apply replacements to the replacement when we create it.  So the loop should no longer be needed.